unsloth/models/cohere.py

```diff
@@ -417,6 +419,10 @@ def CohereModel_fast_forward_inference(
 
     next_decoder_cache = []
     for idx, decoder_layer in enumerate(self.model.layers):
+        decoder_device = decoder_layer.self_attn.q_proj.weight.device
+        hidden_states, out_weight, position_ids = move_to_device(
```

Comment:
wait this move should happen once only - move this outside and put it in the instantiation of the model